### ISSUE TITLE
[Advocacy] [Funding] Create a course to teach data science and AI

### ISSUE THREAD

**Issue by danielinux7** [danielinux7/anana#21]
**Ахцәажәара**

This is about teaching an AI course in [StemLab](http://stemlaboratory.ru/)
Link to [the course](https://github.com/danielinux7/StemLab)

**Ауадаҩрақәа**

The course consists of 24 hours in the period of 3 months.
Questions:
1. What is the goal at the end?
2. What should be included in this course?
3. What platforms should we use?

**Аӡбарақәа**

1. The students at the end of this course should be able to build a machine translator, they should be able to data engineer, to train his/her own NMT model, and utilize statistics in both.
2. Data science should include:
    1. Statistics (random sampling)
    2. Data mining (shell, python)
    3. Data engineering (sentencepiece)
    4. AI model (tensorflow)
    5. Theory
3. [Kaggle](https://www.kaggle.com/), [Collaboratory,](https://colab.research.google.com/) [Huggingface](https://huggingface.co/)

24 lessons:
1: I start off with [AI](https://en.wikipedia.org/wiki/Artificial_intelligence), what is it? Where do we encounter it? What are the possibilities? Then I connect that to [data science](https://en.wikipedia.org/wiki/Data_science), and how data is an essential part, the limitations, I talk about the usage of statistics, computation and understanding of domain. Afterwards I highlight in more detail what we will be covering in this course. Eventually a student should be able to go though the NMT pipeline himself/herself, and build an Abkhazian-Russian machine translator. 
[1.zip](https://github.com/danielinux7/anana/files/8701881/1.zip)

2 & 3: Data mining and preprocess data: show examples with ebooks and web scraping, then build a parallel corpus from a data source using hunalign, bifixer, then practice. ( I need to talk about regex)

4 & 5: Shell scripting and why we use it to manipulate data, I will show them on the Linux terminal, but we will be using Collaboratory, the commands I need to cover shuf, wc, grep, sed, sort, uniq, head, tail. I will give a practical example of a 2000 sentences, remove duplicates, remove numbers, shuffle and show how many lines and words left in the set, then give me the first 200.
Next Lesson, I will give them a task of 5000 sentences to remove duplicates, remove Latin, shuffle and show how many lines and words left in the set, then give me the last 500.

6 & 7: Statistics is used to monitor quality of the data, and AI training, to debug the stats to find any issues. Here we will use random sampling, I will give an explanation what it is and why we use it, then give a live example of 10,000 sentences that have duplicates, getting sampled to 500, by using an excel sheet to rate those sentences. Then I demonstrate how to use a pivot table to debug the stats, to find an anomaly. 
Next Lesson, I give them a task of random sampling a 20,000 sentences (sentences with Latin should be removed) to 500, then collaborate to rate those 500 collectively, I will have a rated file ready, in case they don't make it, afterwards, debug the stats to find the anomaly. 

8 & 9: Data processing (Tokenizing): what is tokenization? the different types, word, subword and character, advantages and disadvantages, then train sentencepiece model in command line with different options and tokenize/detokenize. Next lesson define options to train an sp model, and tokenize/detokenize text.

10: AI models: talk about [machine learning](https://en.wikipedia.org/wiki/Deep_learning), neural networks, talk about pytorch and tensorflow.

11 & 12: NMT model training: Create accounts for students and a space with leaderboard and a testing dataset in HuggingFace and maybe use Kaggle to fine tune a model. and a tokenizer. Next lesson practice.

13: NMT evaluation: using statistics and BLEU score.

14: Features and label: what are those and using tagging, back translation and copy target to enhance NMT quality.
